app/src/routes/project/LayersPanel/Tree/TreeNode.tsx

```diff
@@ -146,21 +156,36 @@ const TreeNode = observer(
                 ) : (
                     <NodeIcon iconClass="w-3 h-3 ml-1 mr-2" node={node.data} />
                 )}
-                <span
-                    className={clsx(
-                        'truncate w-full',
-                        instance
-                            ? selected
-                                ? 'text-purple-100'
-                                : hovered
-                                  ? 'text-purple-200'
-                                  : 'text-purple-300'
-                            : '',
+                <Tooltip>
+                    <TooltipTrigger asChild>
+                        <span
+                            ref={textRef}
+                            className={clsx(
+                                'truncate w-full',
+                                instance
+                                    ? selected
+                                        ? 'text-purple-100'
+                                        : hovered
+                                          ? 'text-purple-200'
+                                          : 'text-purple-300'
+                                    : '',
+                            )}
+                        >
+                            {instance?.component
+                                ? instance.component
+                                : node.data.tagName.toLowerCase()}{' '}
+                            {node.data.textContent}
+                        </span>
+                    </TooltipTrigger>
+                    {isOverflowing && (
+                        <TooltipPortal container={document.getElementById('LayersPanelDiv')}>
```

Comment:
You can use sideOffset to move the tooltip position based on the width of the container. Since the problem is that the node is overflowing, we can calculate the offset as the containerWidth - nodeWidth

https://github.com/onlook-dev/onlook/pull/298/commits/09ce88b6e7cc9ecf0f8c24b1dfaa4ad5190777f2#diff-eced7d84f55e60e45f2876a4fc40b812bb12b845d5b5cdb4d641a619346d96d1R183